lab/hub.go

```diff
@@ -7,16 +7,18 @@ package lab
 import (
 	"context"
 	"errors"
+	progressbar "github.com/cheggaaa/pb/v3"
 	"io"
 	"sync"
 	"sync/atomic"
-
 	"github.com/aau-network-security/haaukins/store"
 	"github.com/aau-network-security/haaukins/virtual/vbox"
 	"github.com/rs/zerolog/log"
 )
 
+
 var (
+	Status   = make(chan string,5)
```

Comment:
This is really dangerous. We have a shared variable that is shared among all clients, hub, events, etc. Imagine a scenario where two clients are creating an event simultaneously, then both will read from the same channel and get the wrong information.

Reply:
I would suggest passing a `context` to the `NewHub` function. When the `daemon` receives a `createEvent` message, it attaches a some kind of logging object to this context. Then the in `NewHub`, this logging object is extracted from the context.

For inspiration, you can look at [this link](https://blog.gopheracademy.com/advent-2016/context-logging/) where they do something similar

Reply:
And to add to this suggestion, you could implement this vaguely described *some kind of  logging object* a new interface that has a `Stream`, `Increment` or `Log` function or something.